app/app/me/components/ProfileEditor.tsx

```diff
@@ -1,142 +1,50 @@
-  university: Yup.string().nullable(),
-  degree_level: Yup.string().nullable(),
-  degree_name: Yup.string().nullable(),
-  degree_semester: Yup.number()
-    .min(1, "Semester must be at least 1.")
-    .integer("Semester must be an integer.")
-    .nullable(),
-  currentJob: Yup.string().nullable(),
-  description: Yup.string().nullable(),
-  job_history: Yup.array().of(
-    Yup.object().shape({
-      employer: Yup.string().required("Employer is required."),
-      position: Yup.string().required("Position is required."),
-      date_from: Yup.date().required("Start date is required."),
-      date_to: Yup.date()
-        .min(Yup.ref("date_from"), "End date must be after the start date.")
-        .required("End date is required."),
-    }),
-  ),
-  social_networks: Yup.array().of(
-    Yup.object().shape({
-      type: Yup.string().required("Type is required."),
-      link: Yup.string()
-        .url("Invalid URL format")
-        .required("Link is required."),
+import {UploadIcon, DownloadIcon} from "@radix-ui/react-icons";
+import { z } from 'zod';
+import { MeModel } from "@models/me";
+
+const profileSchema = z.object({
+  /*
+  profile_picture: z.union([z.string().nullable(), z.instanceof(FileList).optional()])
+    .refine((file) => !file || (file instanceof FileList && file[0].size <= 0.2 * 1048576), {
+      message: "File is too big! Max size is 200 KB.",
     }),
-  ),
+  */
+  first_name: z.string().nonempty("First name is required."),
+  last_name: z.string().nonempty("Last name is required."),
+  birthday: z.date().nullable(),
+  nationality: z.string().nullable(),
+  university: z.string().nullable(),
+  degree_level: z.string().nullable(),
+  degree_name: z.string().nullable(),
+  degree_semester: z.number().min(1, "Semester must be at least 1.").int("Semester must be an integer.").nullable(),
 });
 
-function ProfileEditor({ trigger }) {
-  const { meModel } = useStores();
-  const editorProfile = meModel.editorProfile;
-  const queryClient = useQueryClient();
-
-  const convertImageToBase64 = (file) => {
-    return new Promise((resolve, reject) => {
-      const reader = new FileReader();
-      reader.onload = () => resolve(reader.result);
-      reader.onerror = reject;
-      reader.readAsDataURL(file);
-    });
-  };
-
+function ProfileEditor({ trigger, profile }) {
   return (
     <Dialog trigger={trigger}>
       <Formik
-        initialValues={editorProfile}
-        validationSchema={validationSchema}
-        onSubmit={async (values) => {
-          if (values.profile_picture) {
-            values.profile_picture = await convertImageToBase64(
-              values.profile_picture,
-            );
-          }
-          await meModel.updateEditorProfile(values);
-          await meModel.editProfile();
-          await queryClient.invalidateQueries({ queryKey: ["me"] });
-        }}
+        initialValues={profile}
+        validationSchema={profileSchema}
+        onSubmit={async (values) => {}}
```

Comment:
Maybe add something like this so we don't forget later ;)
```suggestion
        onSubmit={async (values) => {throw Error("TODO: Not implemented - update profile in DB")}}
```

Reply:
Good point, will do! Thanks